### DISCUSSION TITLE
ESPresense battery life?

### DISCUSSION THREAD

**Discussion by ok11** [ESPresense/ESPresense#792] [General]
Hey all,

just wondering if ESPresense works reasonably well with Li-Ion batteries. What I experienced, a device (a pilot one for me) drained a Varta 123A completely in less than 24h. Is that the expecation? I can imagine that continuous polling does not let the device sleep. Is it the case? Would it make sense to try further operating the device with a battery?

TIA!

**Comment by Mapedmaker** [comment]
wondering the same... Did you end up using batteries? I was hoping to use a 10500 mAh power bank

**Reply by ok11** [reply]
Nope, I ended up having a USB connection. Apparently active polling leaves no chance to a battery.

**Reply by Mapedmaker** [reply]
just out of curiosity, how many hours did you get with that battery? i'm just wondering if a power bank could do the job

**Reply by ok11** [reply]
Let's put it this way: a fresh CR123A (1470 mAh) died in one day. I do not know, for how many hours it has run, but for sure less than 24.

**Comment by DTTerastar** [comment]
ESP32's use a decent amount of juice running full tilt.  You don't really see them powered by batteries for that reason.

**Reply by ok11** [reply]
Fair enough. On the other hand, there are tons of instructions around on how to put ESP to deep sleep with a goal to make it more battery-friendly. I just think ESpresense does not have a reliable trigger event to start/stop polling, hence cannot leverage the benefit of ESP sleep.
